### PR TITLE
Updated ServiceCtx usage with adapters

### PR DESCRIPTION
**Pull request by pinebit**
Continuation of https://app.shortcut.com/chainlinklabs/story/30692/migrate-services-to-become-servicectx work.

Implemented using adapters suggested by @jmank88 

Next steps are to update terra, solana, then remove old interfaces and adapters.

### CODE REVIEW

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 6d073d66bb6ef6a2a2f24f57b23d79cb7b13cde7) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/services/job/common.go [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+type ServiceAdapter interface {
+	ServiceCtx
+}
+
+type adapter struct {
+	service Service
+}
+
+// NewServiceAdapter creates an adapter instance for the given Service.
+func NewServiceAdapter(service Service) ServiceCtx {
```
Comment (jmank88) [inline core/services/job/common.go:39]: I wonder if 'adapter' is too generic. Is this more clear?
```suggestion
// NewServiceCtx wraps service to implement ServiceCtx.
func NewServiceCtx(service Service) ServiceCtx {
```
And similarly `s/NewOCR2ProviderAdapter/NewOCR2ProviderCtx`?

Comment (pinebit) [reply core/services/job/common.go:39]: Fixed

**Review by prashantkumar1982 — APPROVED** [review]

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 533c1175769eb4b2e9baaef23fd4e9ca5a4b4c29) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/services/job/common.go (hunk 1 above)
Comment (jmank88) [reply core/services/job/common.go:39]: I see the new ones, but is this old one still needed separately too?

File: core/services/chainlink/application.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ type ChainlinkApplication struct {
 	SessionReaper            utils.SleeperTask
 	shutdownOnce             sync.Once
 	explorerClient           synchronization.ExplorerClient
-	subservices              []interface{} // services.Service or services.ServiceCtx
+	subservices              []services.ServiceCtx
```
Comment (jmank88) [inline core/services/chainlink/application.go:124]: :partying_face:

**Review by jmank88 — APPROVED** [review]

**Pull request merged**